src/plugins/fs/generic.h

```diff
@@ -25,6 +41,7 @@ typedef enum {
     BD_FS_ONLINE_GROW = 1 << 4
 } BDFsResizeFlags;
 
+gboolean bd_fs_can_mkfs (const gchar *type, BDFsMkfsOptionsFlags *options, gchar **required_utility, GError **error);
```

Comment:
I don't like this function. I made it this way because this is how `bd_fs_can_resize` works with `BDFsResizeFlags` but I think it might be too complicated to call a function just to check whether a filesystem supports running mkfs with label. Both UDisks and Blivet need to check this and having the "flags" available directly somewhere might be more useful.

Reply:
Why don't we provide a function that would return the options/flags for the given FS? Instead of calling `bd_fs_can_mkfs()` with all the options/flags.

Reply:
Returning capabilities would be one case, checking if the fs tools are actually installed would be another case. OTOH isn't it better just to fail the actual `bd_fs_mkfs()` call with an _unsupported_ error or similar rather than checking beforehand?

Reply:
We can do both -- fail, but also allow users to check for supported features. And both UDisks and Blivet needs the "check feature" feature so we need it.

Reply:
I think it makes most sense to return the capabilities for the given FS. It can potentially depend on other things than just the tools available (implementation, version of libblockdev, *configure* options,...).